File: python/数组/1710. 卡车上的最大单元数.py
from typing import List
class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        boxTypes.sort(key=lambda x:(x[1],x[0]),reverse=True)
        print(boxTypes)
        res = 0
        for x,y in boxTypes:
            if truckSize>=x:
                res+=x*y
                truckSize-=x
            else:
                res+=truckSize*y
                truckSize=0
            if truckSize == 0:
                break
        return res

s = Solution()
boxTypes = [[5,10],[2,5],[4,7],[3,9]]
truckSize = 10
print(s.maximumUnits(boxTypes,truckSize))

File: go/greedy/1710. 卡车上的最大单元数.go
package greedy

import (
	"fmt"
	"sort"
)

func maximumUnits(boxTypes [][]int, truckSize int) int {
	sort.Slice(boxTypes, func(i, j int) bool {
		return boxTypes[j][1] < boxTypes[i][1]
	})
	fmt.Println(boxTypes)
	var res int
	for _, v := range boxTypes {
		x, y := v[0], v[1]
		if truckSize >= x {
			res += x * y
			truckSize -= x
		} else {
			res += y * truckSize
			truckSize = 0
		}
		if truckSize == 0 {
			break
		}
	}
	return res
}
